### PR TITLE
Faster build

### PR DESCRIPTION
**Pull request by sharwell**
Disable rules which cause severe build-time overhead.

### CODE REVIEW

**Review by mavasani — COMMENTED** [review]
Can you please share more details? We try to run all our analyzers for dogfooding them, and I think that is more critical than build time performance gain.

**Comment by sharwell** [comment]
A rebuild takes 82 seconds prior to this change if VS is not open, and 34 seconds after this change, so it's approximately 2.4x faster. We can enable these rules again one we eliminate the _literal_ 100GiB allocation overhead.

**Comment by sharwell** [comment]
@mavasani Updating Microsoft.CodeAnalysis.NetAnalyzers allows all but two of the analyzers to be enabled again without the performance penalty.

File: eng/Analyzers_ShippingRules.ruleset [hunk 1]
```diff
@@ -32,4 +32,8 @@
     <!-- TODO: Remove this exclusion and fix related issues, tracked by https://github.com/dotnet/roslyn-analyzers/issues/3915 -->
     <Rule Id="RS0041" Action="None" />
   </Rules>
+  <Rules AnalyzerId="Microsoft.CodeAnalysis.NetAnalyzers" RuleNamespace="Microsoft.CodeAnalysis.NetAnalyzers">
+    <Rule Id="CA1508" Action="None" />
```
Comment (mavasani) [inline eng/Analyzers_ShippingRules.ruleset:36]: Can you please file work items to enable them?

**Review by mavasani — APPROVED** [review]

File: eng/Analyzers_ShippingRules.ruleset (hunk 1 above)
Comment (sharwell) [reply eng/Analyzers_ShippingRules.ruleset:36]: Filed #4914 and #4915

**Review by mavasani — APPROVED** [review]

**Pull request merged**